src/background.ts

```diff
@@ -780,6 +801,132 @@ function openEngineDirectory(engineId: string) {
   shell.openPath(path.resolve(engineDirectory));
 }
 
+async function loadVvpp(vvppPath: string) {
+  try {
+    const zipFile = await new Promise<yauzl.ZipFile>((resolve, reject) => {
+      yauzl.open(vvppPath, { lazyEntries: true }, (error, zipFile) => {
+        if (error != null) {
+          reject(error);
+          return;
+        }
+        resolve(zipFile);
+      });
+    });
+    zipFile.readEntry();
+    const manifest: EngineManifest = await new Promise<EngineManifest>(
+      (resolve, reject) => {
+        zipFile.on("entry", async (entry: yauzl.Entry) => {
+          if (entry.fileName !== "engine_manifest.json") {
+            zipFile.readEntry();
+            return;
+          }
+          await new Promise<void>((resolve2, reject2) => {
+            zipFile.openReadStream(entry, async (error, readStream) => {
+              if (error != null) {
+                reject2(error);
+                return;
+              }
+              readStream.on("data", (data) => {
+                resolve(JSON.parse(data.toString()) as EngineManifest);
```

Comment:
`as`はそれ以外の型が来ていた場合、かなり謎のエラーが他で出る形になるので、できればasを使わずにvalidationしつつ型をつけるとメンテナンス性が上がったりします。
ただまあ面倒だと思うので、とりあえずFIXMEコメントを付けておくというのはどうでしょう。
```suggestion
                resolve(JSON.parse(data.toString()) as EngineManifest);// FIXME: asをやめてvalidationする
```

Reply:
確かに。
いつかajvあたりでバリデーションをつけたいですね。